src/compiler/watchPublic.ts

```diff
@@ -419,6 +424,7 @@ namespace ts {
             resolutionCache.finishCachingPerDirectoryResolution();
 
             // Update watches
+            updateExtendedConfigFilePathsWatch(builderProgram.getProgram(), extendedConfigFilesMap || (extendedConfigFilesMap = new Map()), watchExtendedConfigFile);

```

Comment:
This needs to happen wherever `watchConfigFileWildCardDirectories` is called

Reply:
I've refactored this line into `watchExtendedConfigFiles` which is then called after both appearances of `watchConfigFileWildCardDirectories`.